src/systems/battery_plugin/LinearBatteryPlugin.cc

```diff
@@ -333,6 +335,20 @@ void LinearBatteryPlugin::PreUpdate(
           }
         }
       }
+
+      const auto *jointForceCmd =
+        _ecm.Component<components::JointForceCmd>(jointEntity);
+      if (jointForceCmd)
+      {
+        for (double jointForce : jointForceCmd->Data())
+        {
+          if (fabsf(static_cast<float>(jointForce)) > 0)
+          {
+            this->dataPtr->startDraining = true;
```

Comment:
I don't see any logic for stop draining the battery if I stop applying force to the joints. Maybe initializing `this->dataPtr->startDraining` to `false` at `PreUpdate()` makes the trick.

Reply:
I believe that is the intent. There is a trigger to start the battery draining, but we don't monitor the actual forces used by the robot in order to alter how much or how fast the battery drains. The joint velocity command behaves in this manner as well.

Reply:
I think it makes more sense to me to stop draining the battery if there are no velocity or force commands but if that's the intent the code looks good to me. We can revisit it later if needed.

Reply:
How about tackling this feature under the umbrella of your battery recharging work.

Reply:
Sounds good!